tests/unit/delegate.ts

```diff
@@ -0,0 +1,180 @@
+import registerSuite = require('intern!object');
+import assert = require('intern/chai!assert');
+import { emit } from 'dojo-core/on';
+import { Handle } from 'dojo-core/interfaces';
+import delegate from 'src/delegate';
+import * as dom from 'src/dom';
+
+let container = document.createElement('div');
+let handles: (Handle)[] = [];
+
+function testDelegate(...args: any[]) {
```

Comment:
About 5 tests weren't calling this function (they were calling `delegate` directly and thus their handles were never cleaned up), but it's also trivial to wrap this around each delegate call (i.e. `handles.push(delegate(...))`) so I'll be doing that instead.